tests/integration/sync/test_sync_watch.py

```diff
@@ -0,0 +1,251 @@
+import os
+import shutil
+import time
+import uuid
+
+import logging
+import json
+import tempfile
+from pathlib import Path
+from unittest import skipIf
+
+import boto3
+from botocore.config import Config
+from parameterized import parameterized
+
+from samcli.lib.bootstrap.bootstrap import SAM_CLI_STACK_NAME
+from samcli.lib.utils.resources import (
+    AWS_APIGATEWAY_RESTAPI,
+    AWS_LAMBDA_FUNCTION,
+    AWS_STEPFUNCTIONS_STATEMACHINE,
+)
+from tests.integration.buildcmd.build_integ_base import BuildIntegBase
+from tests.integration.sync.sync_integ_base import SyncIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
+from tests.integration.sync.test_sync_code import API_SLEEP, SFN_SLEEP
+
+from tests.testing_utils import (
+    RUNNING_ON_CI,
+    RUNNING_TEST_FOR_MASTER_ON_CI,
+    RUN_BY_CANARY,
+    kill_process,
+    read_until_string,
+    start_persistent_process,
+)
+from tests.testing_utils import run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_SYNC_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+
+LOG = logging.getLogger(__name__)
+
+LOG.handlers = []  # This is the key thing for the question!
+
+# Start defining and assigning your handlers here
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(message)s")
+handler.setFormatter(formatter)
+LOG.addHandler(handler)
+
+
+@skipIf(SKIP_SYNC_TESTS, "Skip sync tests in CI/CD only")
+class TestSyncWatch(SyncIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        PackageIntegBase.setUpClass()
+        cls.test_data_path = Path(__file__).resolve().parents[1].joinpath("testdata", "sync")
+
+    def setUp(self):
+        self.cfn_client = boto3.client("cloudformation")
+        self.ecr_client = boto3.client("ecr")
+        self.lambda_client = boto3.client("lambda")
+        self.api_client = boto3.client("apigateway")
+        self.sfn_client = boto3.client("stepfunctions")
+        self.stacks = []
+        self.s3_prefix = uuid.uuid4().hex
+        self.test_dir = Path(tempfile.mkdtemp())
+        # Remove temp dir so that shutil.copytree will not throw an error
+        # Needed for python 3.6 and 3.7 as these versions don't have dirs_exist_ok
+        shutil.rmtree(self.test_dir)
+        shutil.copytree(self.test_data_path, self.test_dir)
```

Comment:
Since we are only using one version in the temp dir, is copying the whole test_data_path content necessary? I think update_file kind of makes the copying action redundant, I would only consider copying one set

Reply:
Oh nevermind I think having both copied is useful for infra watch where the template is replaced. Having both would make the template switch smoother. You can ignore this comment

Reply:
Oh nevermind I think having both copied is useful for infra watch where the template is replaced. Having both would make the template switch smoother. You can ignore this comment